### PR TITLE
[ADP-3152] Export of FineTypes to Haskell

### PR DESCRIPTION
**Pull request by HeinrichApfelmus**
This pull requests implements the export of FineTypes `Typ` to Haskell types.

- [x] The command `fine-types convert` is able to export a FineTypes module to a Haskell module.
- [x] The types in the exported Haskell module are instances of a class `ToValue a` with member function `toValue :: a -> Value`.
- [x] A unit test covers the functionality by autogenerating a Haskell module from the file and using it in the `fine-types-test` package.

### CODE REVIEW

File: lib/fine-types-test/Setup.hs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+main = defaultMainWithHooks hooks
+
+hooks :: UserHooks
+hooks = simpleUserHooks{postConf = \_ _ _ lbi -> generateSources lbi}
+
+generateSources :: LocalBuildInfo -> IO ()
+generateSources lbi =
+    forM_ (allComponentsInBuildOrder lbi) $ \clbi -> do
+        let dir = autogenComponentModulesDir lbi clbi
+            file = dir </> "Language/FineTypes/Test/UTxO.hs"
```
Comment (HeinrichApfelmus) [inline lib/fine-types-test/Setup.hs:19]: Todo: Make this more general, work with any `LocalBuildInfo`, not just this particular file.

File: lib/fine-types-test/Setup.hs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        putStrLn $ concat ["*** Creating ", file, "."]
+        createDirectoryIfMissing True (takeDirectory file)
+        callProcess
+            "fine-types"
+            [ "convert"
+            , "--haskell"
```
Comment (HeinrichApfelmus) [inline lib/fine-types-test/Setup.hs:25]:
```
finite-type convert haskell
finite-type convert json --format yaml
```

File: lib/fine-types-test/Setup.hs (hunk 1 above)
Comment (HeinrichApfelmus) [reply lib/fine-types-test/Setup.hs:19]: To be done in a separate pull request. ADP-3180

**Review by paolino — APPROVED** [review]
LGTM, great milestone :-)

**Comment by HeinrichApfelmus** [comment]
Thanks! 😊

**Pull request merged**